src/app/components/OrderInput.tsx

```diff
@@ -805,10 +876,184 @@ function InputTooltip({ message }: { message: string }) {
 }
 
 // TODO(dcts): implement percentage slider in future PR
-function PercentageSlider() {
-  return <></>;
+interface PercentageSliderProps {
+  initialPercentage: number;
+  callbackOnPercentageUpdate: (newPercentage: number) => void;
+  isLimitOrder: boolean;
+  isBuyOrder: boolean;
+  isSellOrder: boolean;
 }
 
+const PercentageSlider: React.FC<PercentageSliderProps> = ({
+  initialPercentage,
+  callbackOnPercentageUpdate,
+  isLimitOrder,
+  isBuyOrder,
+  isSellOrder,
+}) => {
+  const [percentage, setPercentage] = useState(initialPercentage);
+  const [toolTipVisible, setToolTipVisible] = useState(false);
+  const sliderRef = useRef<HTMLInputElement>(null);
+  const { token1, token2 } = useAppSelector((state) => state.orderInput);
+
+  const inputToken1 = useAppSelector((state) => state.orderInput.token1.amount);
+  const inputToken2 = useAppSelector((state) => state.orderInput.token2.amount);
+
+  const balanceToken1 =
+    useAppSelector((state) => selectBalanceByAddress(state, token1.address)) ||
+    0;
+  const balanceToken2 =
+    useAppSelector((state) => selectBalanceByAddress(state, token2.address)) ||
+    0;
+
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const newPercentage = parseInt(e.target.value, 10);
+    setPercentage(newPercentage);
+    callbackOnPercentageUpdate(newPercentage);
+  };
+
+  useEffect(() => {
+    if (!sliderRef.current) {
+      return;
+    }
+
+    sliderRef.current.value = "0";
+    sliderRef.current.style.backgroundSize = `0% 100%`;
+    setPercentage(0);
+
+    if (inputToken1 > 0 && isLimitOrder && isBuyOrder) {
+      return;
+    } else if (inputToken2 > 0 && isLimitOrder && isSellOrder) {
+      return;
+    } else if (balanceToken2 && inputToken2 > 0) {
+      const newPercentage = Calculator.multiply(
+        Calculator.divide(inputToken2, balanceToken2),
+        100
+      );
+      setPercentage(newPercentage);
+      sliderRef.current.style.backgroundSize = `${newPercentage}% 100%`;
+    } else if (balanceToken1 && inputToken1 > 0) {
+      const newPercentage = Calculator.multiply(
+        Calculator.divide(inputToken1, balanceToken1),
+        100
+      );
+      setPercentage(newPercentage);
+      sliderRef.current.style.backgroundSize = `${newPercentage}% 100%`;
+    }
+  }, [
+    inputToken1,
+    balanceToken1,
+    inputToken2,
+    balanceToken2,
+    isLimitOrder,
+    isBuyOrder,
+    isSellOrder,
+  ]);
+
+  return (
+    <>
+      <div className="slider-container rounded-md w-full relative mt-5">
```

Comment:
can you make the full slider opacity of 0.7, currently there is too much emphasis on it.

```suggestion
      <div className="slider-container rounded-md w-full relative mt-5 opacity-70">
```